src/sink/tcp.rs

```diff
@@ -39,17 +52,25 @@ pub struct Tcp {
 pub struct Config {
     pub host: String,
     pub port: u16,
-    #[serde(default = "ttl")]
+    #[serde(default = "default_ttl")]
     pub ttl: u32,
-    #[serde(default = "t")]
+    #[serde(default = "default_no_delay")]
     pub is_no_delay: bool,
+    #[serde(with = "either::serde_untagged_optional")]
```

Comment:
In order to allow this to be optional, i think you need to do:

```suggestion
    #[serde(with = "either::serde_untagged_optional", default = "Default::default")]
```